panel/param.py

```diff
@@ -357,8 +358,10 @@ def _get_widgets(self):
         filtered = [(k,p) for (k,p) in sorted_precedence
                     if ((p.precedence is None) or (p.precedence >= self.display_threshold))]
         groups = itertools.groupby(filtered, key=key_fn)
-        sorted_groups = [sorted(grp) for (k,grp) in groups]
-        ordered_params = [el[0] for group in sorted_groups for el in group]
+        # Params preserve definition order in Python 3.6+
+        ordered_params = sys.version_info.major == 3 and sys.version_info.minor >= 6
```

Comment:
Since Guido says Python 4.0 will be the version after 3.9, let's include >=4.x here.

Reply:
Fixed in ec923e7